src/main/cpp/Robot.cpp

```diff
@@ -1,221 +1,203 @@
 #include "Robot.h"
-#include <iostream>
+
 #include <frc/smartdashboard/SmartDashboard.h>
 
+#include <iostream>
 
-void 
-Robot::RobotInit() {
-  m_chooser.SetDefaultOption("Blue", blueAlliance);
-  m_chooser.AddOption("RED", redAlliance);
-  frc::SmartDashboard::PutData("Alliance Color", &m_chooser);
-  
-  camera = frc::CameraServer::GetInstance()->StartAutomaticCapture();
-  
-  try{
-    navx = new AHRS(frc::SPI::Port::kMXP);
-  } catch(const std::exception& e){
-    std::cout << e.what() <<std::endl;
-  }
-  m_swerve.debug(*navx);
-  m_climbing = false;
-}
+void Robot::RobotInit() {
+    m_chooser.SetDefaultOption("Blue", blueAlliance);
+    m_chooser.AddOption("RED", redAlliance);
+    frc::SmartDashboard::PutData("Alliance Color", &m_chooser);
 
+    frc::CameraServer::StartAutomaticCapture();
 
-void 
-Robot::RobotPeriodic() {
+    try {
+        navx = new AHRS(frc::SPI::Port::kMXP);
+    } catch (const std::exception &e) {
+        std::cout << e.what() << std::endl;
+    }
+    m_swerve.debug(*navx);
+    m_climbing = false;
 }
 
-
-void 
-Robot::AutonomousInit() {
-  m_auto.ResetAuto();
-  m_time = 0;
-  m_swerve.ResetOdometry();
-  m_swerve.GenerateTrajectory_1();
-  m_swerve.Initialize();
-  m_intake.Deploy();
-  m_shooter.setState(Shooter::State::IDLE);
-  m_intake.setState(Intake::State::IDLE);
-  m_shooter.Zero();
-  PDH.ClearStickyFaults();
-  navx->Reset();
-
-  //This might not work??
-  m_climber.Initialize();
+double Robot::GetGyroAngleRad() {
+    if (navx) {
+        // flip direction, navx is CW around Z, not CCW
+        // but double flip since roborio is upsidedown
+        return navx->GetYaw() * (M_PI / 180.0);
+    }
+    return 0.0;
 }
 
-
-void 
-Robot::AutonomousPeriodic() {
-  m_time += m_timeStep;
-  m_auto.Periodic(m_time);
-  //frc::SmartDashboard::PutNumber("Y", m_swerve.GetYPosition());
-  //frc::SmartDashboard::PutNumber("X", m_swerve.GetXPostion());
-
-  switch(m_auto.getState()){
-    case AutoMode::State::IDLE:
-      m_intake.setState(Intake::IDLE);
-      m_shooter.setState(Shooter::IDLE);
-      m_swerve.Drive(0, 0, 0, 0, true);
-      break;
-    case AutoMode::State::DRIVEnINTAKE:
-      m_intake.setState(Intake::State::RUN);
-      m_shooter.setState(Shooter::State::LOAD);
-      m_swerve.TrajectoryFollow(navx->GetYaw(), m_auto.getWaypointIndex());
-      break;
-    case AutoMode::State::SHOOT:
-      m_shooter.setState(Shooter::State::SHOOT);
-      break;
-    case AutoMode::State::DRIVE:
-      m_swerve.TrajectoryFollow(navx->GetYaw(), m_auto.getWaypointIndex());
-      break;
-    case AutoMode::State::INTAKE:
-      // m_intake.setState(Intake::State::RUN);
-      // m_shooter.setState(Shooter::State::LOAD);
-      break;
-    default:
-      break;
-  }
-
-  m_intake.Periodic();
-  m_shooter.Periodic(m_swerve.GetXPosition(), m_swerve.GetYPosition());
+void Robot::RobotPeriodic() {
+    m_swerve.UpdateOdometry(GetGyroAngleRad(), m_timeStep);
```

Comment:
Call update odom in RobotPeriodic instead of only when following trajectories, this sets you up for shooter compensation later.